### PR TITLE
Audio system improvement

### PR DESCRIPTION
**Pull request by XorJS**
**Changed**
- Added: Data-driven for audio clips.
- Added: Editor/Run time validation for audio clips (size/tag).
- Added: Editor/Run time validation for musics (size/tag).
- Removed: Unused variable (m_lastWorldMapMusicTime)

**Preview**
image

### CODE REVIEW

File: unity/Assets/Scripts/Engine/Audio/AudioManager.cs [hunk 1]
```diff
@@ -46,6 +56,35 @@ protected override void Awake()
         base.Awake();
     }
 
+    void OnValidate()
```
Comment (robinlavallee) [inline unity/Assets/Scripts/Engine/Audio/AudioManager.cs]: Who invokes this method? Is it done by the Editor?

File: unity/Assets/Scripts/Engine/Audio/AudioManager.cs [hunk 2]
```diff
@@ -83,41 +122,23 @@ public float SFXVolume
         }
     }
 
-    public void PlayDig()
-    {
-        PlayAudioClip(m_digAudio);
-    }
-
-    public void PlayCut()
+    public void PlaySound(E_Sound sound)
```
Comment (robinlavallee) [inline unity/Assets/Scripts/Engine/Audio/AudioManager.cs]: In the future, we will probably want to refer to sounds by strings instead, to allow complete customization from the users.

Comment (XorJS) [reply unity/Assets/Scripts/Engine/Audio/AudioManager.cs]: https://docs.unity3d.com/ScriptReference/MonoBehaviour.OnValidate.html

"This function is called when the script is loaded or a value is changed in the Inspector (Called in the editor only).

You can use this to ensure that when you modify data in an editor, that data stays within a certain range."

File: unity/Assets/Scripts/Engine/Audio/AudioManager.cs (hunk 1 above)
Comment (XorJS) [reply unity/Assets/Scripts/Engine/Audio/AudioManager.cs]: https://docs.unity3d.com/ScriptReference/MonoBehaviour.OnValidate.html

"This function is called when the script is loaded or a value is changed in the Inspector (Called in the editor only).

You can use this to ensure that when you modify data in an editor, that data stays within a certain range."

**Pull request merged**